### PR TITLE
[JENKINS-35016] Migrate to new parent POM

### PR DESCRIPTION
**Pull request by andresrc**
See [JENKINS-35016](https://issues.jenkins-ci.org/browse/JENKINS-35016)

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,16 +34,20 @@
     <url>http://github.com/jenkinsci/aws-java-sdk-plugin</url>
     <tag>HEAD</tag>
   </scm>
+  <properties>
+    <jenkins.version>1.580.3</jenkins.version>
```
Comment (andresrc) [inline pom.xml]: Jackson dependency already brought 1.580

File: pom.xml [hunk 2]
```diff
@@ -34,16 +34,20 @@
...
+    <java.level>6</java.level>
```
Comment (andresrc) [inline pom.xml]: Level bump motivated by both baseline bump and AWS Java SDK dependency.

File: pom.xml [hunk 3]
```diff
@@ -69,14 +73,11 @@
       <artifactId>jackson2-api</artifactId>
       <version>2.5.4</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpclient</artifactId>
+      <version>4.5</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (andresrc) [inline pom.xml]: To avoid `InjectedTests` conflict.

**Comment by armfergom** [comment]
LGTM 🐝

File: pom.xml (hunk 1 above)
Comment (kwhetstone) [inline pom.xml]: W

**Comment by kwhetstone** [comment]
:bee:

**Comment by francisu** [comment]
Who is the maintainer for this one? LGTM as well. Should I merge it?

**Comment by andresrc** [comment]
@ndeloof and/or @Vlatombe ok to merge and release?

**Comment by ndeloof** [comment]
This is a lib bundle plugin, I don't think there's a maintainer for it, or there's even need to define one. Anybody with spare time is welcome to maintain it ;)

**Comment by Vlatombe** [comment]
Seem to have caused https://issues.jenkins-ci.org/browse/JENKINS-35242 ?

**Comment by ndeloof** [comment]
omg. Please offer Jenkins a plugin mechanism with descent classloader isolation :'(

**Pull request merged**